rust/arrow/src/record_batch.rs

```diff
@@ -103,6 +103,56 @@ impl RecordBatch {
         RecordBatch { schema, columns }
     }
 
+    /// Creates a new [`RecordBatch`] with no columns
+    ///
+    /// TODO add an code example using `append`
+    pub fn new() -> Self {
```

Comment:
I think it's worthwhile to maintain the property that a recordbatch is ummutable. So, this feels a bit out of place for me.

How about we create a builder that takes an iterator of `(&str, ArrayRef)` instead?

The below is what I mean by out of place

```rust
let mut batch = read_batch_from_somewhere()?;
batch = batch.append("field", array_ref)?;
```

There might be a benefit in having the above, so I'm open to convincing :)
This could partially address your question on the TODO about returning `Self` on error

Reply:
This makes sense @nevi-me  -- I think a separate `RecordBatchBuilder` sounds like a better idea to me. I'll wait a bit for feedback from others, and then give it a shot.